doc/api/util.md

```diff
@@ -1097,6 +1097,80 @@ console.log(values, positionals);
 // Prints: [Object: null prototype] { foo: true, bar: 'b' } []
 ```
 
+Detailed parse information is available for adding custom behaviours by
+specifying `tokens: true` in the configuration. The returned tokens have
+properties describing:
+
+* all tokens
+  * `kind` { string } One of 'option', 'positional', or 'option-terminator'.
+  * `index` { number } Index of element in `args` containing token.
+* option tokens
+  * `name` { string } Long name of option.
+  * `rawName` { string } How option used in args, like `-f` of `--foo`.
+  * `value` { string | undefined } Option value specified in args.
+    Undefined for boolean options.
+  * `inlineValue` { boolean | undefined } Whether option value specified inline,
+    like `--foo=bar`.
+* positional tokens
+  * `value` { string } Positional value (i.e. `args[index]`).
+* option-terminator token
+
+For example, assuming the following script which uses
+automatic detection of options and no error checking:
+
+```mjs
+import { parseArgs } from 'node:util';
+console.log(parseArgs({ strict: false, tokens: true }));
+```
+
+```cjs
+const { parseArgs } = require('node:util');
```

Comment:
Rather than a code snippet that demonstrates the three types of tokens, I think it would be more valuable to pick a real world use-case that would benefit from having parse information.

Perhaps the [stopping at the first positional](https://github.com/pkgjs/parseargs/issues/125) of @bakkot's?

Is there a better use case that's a bit less clunky?

Reply:
> Is there a better use case that's a bit less clunky?

[Handling --no-foo](https://github.com/pkgjs/parseargs/blob/b1b632c669662e8f54035507311fddbbbea5f674/examples/negate.js), maybe?

Reply:
I'll have a go at a zero-config negate.

Reply:
negate sounds great to me.

Reply:
Added negate as example. I left it as strict with declared options to make it clear that the negative option needs to be declared (which I think @bakkot mentioned at some point), but the example usage would work the same with `strict:false` and no declared options.